llvm/lib/Target/RISCV/RISCVTargetTransformInfo.cpp

```diff
@@ -1460,8 +1460,29 @@ InstructionCost RISCVTTIImpl::getVectorInstrCost(unsigned Opcode, Type *Val,
   std::pair<InstructionCost, MVT> LT = getTypeLegalizationCost(Val);
 
   // This type is legalized to a scalar type.
-  if (!LT.second.isVector())
-    return 0;
+  if (!LT.second.isVector()) {
+    auto *FixedVecTy = dyn_cast<FixedVectorType>(Val);
+    // Scalable vectors can't be legalized with scalars, return invalid cost.
+    if (!FixedVecTy)
```

Comment:
Does this bit actually get hit?  I'd have expected it to go through the  if (LT.second.isScalableVector() && !LT.first.isValid()) check just below.

Reply:
Yes, the condition below handles it, removed this